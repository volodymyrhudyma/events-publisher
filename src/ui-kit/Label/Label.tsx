import React, { ReactNode } from 'react';
import { Wrapper } from './styled/Label.styled';

interface ILabelProps {
  children: ReactNode;
  uppercase: boolean;
  color: string;
  /* tslint:disable-next-line */
  onClick?: (value: any) => void;
}

const Label = ({ uppercase, color, children, onClick }: ILabelProps) => (
  <Wrapper uppercase={uppercase} color={color} onClick={onClick}>
    {children}
  </Wrapper>
);

Label.defaultProps = {
  uppercase: true,
  color: 'primary',
};

export default Label;
