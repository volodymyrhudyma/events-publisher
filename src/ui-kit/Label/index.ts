import Component from './Label';

export default Component;
