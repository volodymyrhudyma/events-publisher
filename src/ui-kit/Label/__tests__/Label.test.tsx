import React from 'react';
import { shallow } from 'enzyme';
import Label from '../Label';

describe('Label component', () => {
  it('should render correctly', () => {
    const component = shallow(<Label>Label</Label>);
    expect(component).toMatchSnapshot();
  });
});
