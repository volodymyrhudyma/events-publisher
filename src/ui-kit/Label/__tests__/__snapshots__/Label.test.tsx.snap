// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should render correctly 1`] = `
<styled.label
  color="primary"
  uppercase={true}
>
  Label
</styled.label>
`;
